### PR TITLE
Refactor with updated validators

### PR DESCRIPTION
**Pull request by kodumbeats**
## What does this PR do?

In the related PR, Whitelist validators were given `type=string` by default. This PR defines the type of any whitelist that doesn't exhibit the new default behavior.

This PR should remain as a draft until the related PR is deployed to a version tag.

## Test Plan

The related Utopia PR covers unit testing

## Related PRs and Issues

https://github.com/utopia-php/framework/pull/6

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

### CODE REVIEW

File: app/controllers/api/users.php [hunk 1]
```diff
@@ -346,7 +346,7 @@
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_USER)
     ->param('userId', '', new UID(), 'User unique ID.')
-    ->param('status', '', new WhiteList([Auth::USER_STATUS_ACTIVATED, Auth::USER_STATUS_BLOCKED, Auth::USER_STATUS_UNACTIVATED], true), 'User Status code. To activate the user pass '.Auth::USER_STATUS_ACTIVATED.', to block the user pass '.Auth::USER_STATUS_BLOCKED.' and for disabling the user pass '.Auth::USER_STATUS_UNACTIVATED)
+    ->param('status', '', new WhiteList([Auth::USER_STATUS_ACTIVATED, Auth::USER_STATUS_BLOCKED, Auth::USER_STATUS_UNACTIVATED], true, 'integer'), 'User Status code. To activate the user pass '.Auth::USER_STATUS_ACTIVATED.', to block the user pass '.Auth::USER_STATUS_BLOCKED.' and for disabling the user pass '.Auth::USER_STATUS_UNACTIVATED)
```
Comment (christyjacob4) [inline app/controllers/api/users.php:349]: Would it be a good idea to replace this with Model::TYPE_INTEGER

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please take a look at the coment

File: app/controllers/api/users.php (hunk 1 above)
Comment (kodumbeats) [reply app/controllers/api/users.php:349]: We could use the new definitions from Utopia when they're merged?
https://github.com/utopia-php/framework/pull/6/files#diff-9dad0f0dabc730596b3399e6fe25aed903dba8bc7836fe778fce387cf9f3e111R19

Comment (christyjacob4) [reply app/controllers/api/users.php:349]: Ah yes!! That's better :)

**Comment by eldadfux** [comment]
Please sync 0.8/dev for tests to pass

**Comment by eldadfux** [comment]
@kodumbeats Do you have changes required for this PR in the latest utopia/framework release ready?

**Comment by kodumbeats** [comment]
@eldadfux 
Torsten opened a PR against this branch to make the whitelist changes he needed for spec generation:
https://github.com/appwrite/appwrite/pull/1087

That PR pulls the latest `utopia/framework` into Appwrite so I can make the final changes, but it depends on a `utopia/storage` change, which is ready: https://github.com/utopia-php/storage/pull/14

I've reviewed both of these PRs, so once we start merging, I can make the quick changes here and 🚢 everything together.

**Comment by eldadfux** [comment]
utopia/storage changes are available now in v0.5

**Comment by kodumbeats** [comment]
Opening so it can be merged once we have 🟢🟢🟢

**Comment by eldadfux** [comment]
@kodumbeats tests are failing because of the missing `Mock` validator

**Review by christyjacob4 — APPROVED** [review]
Looks Good

**Pull request merged**